src/app/services/trips.service.ts

```diff
@@ -110,10 +111,14 @@ export class TripsService {
   }
 
 
-  getTrip(tripIndex: string) {
+  getTrip(tripIndex: string): Trip {
     return this.trips.filter((trip) => {
```

Comment:
@ashish173  currently i have not implemented trip-detail with the use of store. so its doesn't return an observable. I will change it when it will be implemented that way.